plugins/libplugin-pay.c

```diff
@@ -251,16 +251,38 @@ static void channel_hints_update(struct payment *p,
 		struct channel_hint *hint = &root->channel_hints[i];
 		if (short_channel_id_eq(&hint->scid.scid, &scid) &&
 		    hint->scid.dir == direction) {
+			bool modified = false;
 			/* Prefer to disable a channel. */
-			hint->enabled = hint->enabled & enabled;
+			if (!enabled && hint->enabled) {
+				hint->enabled = false;
+				modified = true;
+			}
 
 			/* Prefer the more conservative estimate. */
 			if (estimated_capacity != NULL &&
 			    amount_msat_greater(hint->estimated_capacity,
-						*estimated_capacity))
+						*estimated_capacity)) {
 				hint->estimated_capacity = *estimated_capacity;
-			if (htlc_budget != NULL && *htlc_budget < hint->htlc_budget)
+				modified = true;
+			}
+			if (htlc_budget != NULL && *htlc_budget < hint->htlc_budget) {
 				hint->htlc_budget = *htlc_budget;
+				modified = true;
+			}
+
+			if (!modified)
+				/* Do not print a log.  */
+				return;
+
+			paymod_log(p, LOG_DBG,
+				   "Updated a channel hint for %s: "
+				   "enabled %s, estimated capacity %s",
+				   type_to_string(tmpctx,
+						  struct short_channel_id_dir,
+						  &hint->scid),
+				   hint->enabled ? "true" : "false",
+				   type_to_string(tmpctx, struct amount_msat,
+						  &hint->estimated_capacity));
```

Comment:
imho it's preferable to have the non-negated `if` surround the log, rather than having an early exit in the form of that `return`